src/Analyzers/Core/Analyzers/MakeFieldReadonly/MakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -104,19 +107,20 @@ void OnSymbolEnd(SymbolAnalysisContext symbolEndContext)
                     }
                 }
 
-                static bool IsCandidateField(IFieldSymbol symbol) =>
+                static bool IsCandidateField(IFieldSymbol symbol, INamedTypeSymbol threadStaticAttribute) =>
                         symbol.DeclaredAccessibility == Accessibility.Private &&
                         !symbol.IsReadOnly &&
                         !symbol.IsConst &&
                         !symbol.IsImplicitlyDeclared &&
                         symbol.Locations.Length == 1 &&
                         symbol.Type.IsMutableValueType() == false &&
-                        !symbol.IsFixedSizeBuffer;
+                        !symbol.IsFixedSizeBuffer &&
+                        !symbol.GetAttributes().Any(a => SymbolEqualityComparer.Default.Equals(a.AttributeClass, threadStaticAttribute));
```

Comment:
this will allocate on each field.  can you write a helper to do this with an explicit loop and no lambda?

Reply:
💡 You can also use a static lambda, passing the state as a separate argument to the call.

Reply:
➡️ Moved my comment to https://github.com/dotnet/roslyn/pull/48361#discussion_r500611678